クレジットカード登録画面

zerotaisuke opened the pull request:
# what
・クレジットカード登録画面のビューを作成。
・各種クレジットロゴは、images/credit_logoに保存してimageタグで取得。
・専用scssファイル（credit1.scss)を作成。

Commit:
Merge remote-tracking branch 'ryo-kamido/master' into クレジットカード登録画面

Commit:
クレジットカード登録画面ビュー

# what
クレジットカード登録画面のビューを作成
各種クレジットロゴは、images/credit_logoに保存してimageタグで取得。

yuki-uchida commented on app/assets/stylesheets/credit1.scss:
```diff
@@ -0,0 +1,140 @@
+
+
```
credit1という名前は不適切です。
1が何を表しているのか分からないためです。
他の名前を検討してください

zerotaisuke replied:
registration_creditcard
に変更しました。

yuki-uchida commented on app/assets/stylesheets/credit1.scss:
```diff
@@ -0,0 +1,140 @@
+  .form-group.signup-form-expire{
+    width: 100%;
+  .select-wrap1{
+      display: inline-block;
```
この辺り、BEMに沿ったクラス名が付けられていません。こちらを参考に、アンバサンドを活用した記述にしてくださいhttps://qiita.com/Takuan_Oishii/items/0f0d2c5dc33a9b2d9cb1

zerotaisuke replied:
アンバサンドを活用した記述に修正しました

yuki-uchida commented on app/views/profiles/credit1.html.haml:
```diff
@@ -0,0 +1,74 @@
+          .select-wrap1
+            %select#payment_card_expire_mm.select-default
+              %option{:value => "01"} 01
+              %option{:value => "02"} 02
```
railsのdate selectで良いのでは？
一度調べてみてください

zerotaisuke replied:
調べました。
今回においてはoptionで進めさせていただけますと幸いです。

yuki-uchida replied:
それはなぜですか？

zerotaisuke replied:
ご指摘をいただいたのちにdeta_selectに関する記事を読み漁り、年と月をdate_selectで表現することは可能と感じましたが、「http://mooooooooooriiiiii.hatenablog.com/entry/2017/05/15/180405」
の記事にて、deta_selectで出力される値は、YYYY/MM/01となり、日付の値は強制的に「1日」が出力されるとのことでした。
　ここからは、私個人の能力の問題になりますが、この「1日」という日付が今後、どういう影響を与えられるかの想像がつかない状況です。加えて当該箇所は、日付の受け渡しを行うのであれば、月初の日付ではなく月末の日付を受けわたすことが正確であるため、仮に日付の値を受けわたすのであれば、「31」の値にしなければならないと考えます。
　deta_selectについて理解を深めることで上記問題を解決できることも予想されますが、最終課題の期限の問題もあるため、他の実装を優先させて、改めて理解を深めたいと考えています。
以上のことから、クレジットカードという情報の重要性を鑑みて、不甲斐ないですが私の能力で確実に表現ができ、正確な情報を受けわたすことができるであろう、oputionの使用で進めたいと思っております。
以上、ご確認のほどよろしくお願いいたします。

yuki-uchida replied:
かしこまりました一応対策もお伝えしておきますね


日付の部分だけ追加で選択し、送信しRails側で結合すれば良いでしょう。

期限もあるため、今回はこのまま行きましょう

zerotaisuke replied:
恐れ入ります。ありがとうございます。
後日にはなってしまいますが、ご指導いただきました点をはじめさらに理解を深めることができるように努力いたします。

zerotaisuke commented:
<img width="1440" alt="スクリーンショット 2019-03-16 17 48 31" src="https://user-images.githubusercontent.com/47076525/54472960-b2ee0580-4814-11e9-9e9d-7333615f0a92.png">

zerotaisuke commented:
前回プレビュー画面を添付し忘れておりました。大変失礼いたしました。
現状のプレビュイー画面を添付いたします。

Commit:
ファイル名の変更およびBem記述に変更

# what
アンバサンドを活用した記述に修正
当初のファイル名「credit1」を「registration_creditcard」に変更
ファイル名変更に伴い、routes.rb、controllerの修正

yuki-uchida commented on app/assets/stylesheets/registration_creditcard.scss:
```diff
@@ -0,0 +1,145 @@
+    width: 34px;
+  }
+
+  .masterlogo{
```
この辺りも全てBEMに沿ってクラス命名をしてください。
もし様々なところで使われるCSSの場合にはmixinで呼び出す形にしましょう

zerotaisuke replied:
# what
・ロゴ部分の記述をBEMに沿って命名をしました。
・別途登録されていたmixinの記述を活用して表現するようにしました。
ｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰ
@mixin height_width($height,$width){
  height: $height;
  width:  $width;
}
ｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰ
・コメントアウトの記述を行い、メンテナンスしやすくしました。

Commit:
BEMの修正

# what
・ロゴ部分の記述をBEMに沿って命名をしました。
・別途登録されていたmixinの記述を活用して表現するようにしました。
ｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰ
@mixin height_width($height,$width){
  height: $height;
  width:  $width;
}
ｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰｰ
・コメントアウトの記述を行い、メンテナンスしやすくしました。

yuki-uchida commented on app/assets/stylesheets/registration_creditcard.scss:
```diff
@@ -0,0 +1,142 @@
+  background-color: white;
+}
+
+.l-chapter-credithead{
```
この辺りもBEMに沿ってください。全体的に見直してみてください

zerotaisuke replied:
# what
メルカリに合わせていたクラス名をBEM記法に合わせて変更

Commit:
registration_creditcardのクラス名とscssを修正

# what
メルカリに合わせていたクラス名をBEM記法に合わせて変更

yuki-uchida reviewed (approved):
LGTMです。お疲れ様でした

zerotaisuke commented:
ありがとうございました。引き続きご指導のほどよろしくお願いいたします。

ryo-kamido committed:
Merge branch 'master' into クレジットカード登録画面
